src/main/scala/karazin/scala/users/group/week2/homework/adt.scala

```diff
@@ -38,29 +43,46 @@ object adt:
       Make sure that in case of failing the method with exception
       no exception is thrown but the case for an error is returned
      */
-    def map = ???
-  
+    def map[Q](f: V ⇒ Q): ErrorOr[Q] =
+      this match
+        case ErrorOr.Error(v) ⇒ ErrorOr.Error(v)
+        case ErrorOr.Value(v)  ⇒ try ErrorOr.Value(f(v)) catch{
+          case ex: Throwable => ErrorOr.Error(ex)
+        }
+      
     /* 
       The method is used for filtering
       Provide a type parameter, an argument and a result type
       
       Make sure that in case of failing the method with exception
       no exception is thrown but the case for an error is returned
      */
-    def withFilter = ???
-  
+    def withFilter(p: V => Boolean): ErrorOr[V] =
```

Comment:
You don't need to fix it. But. This method is a very good example that not all boxes could have meaningful withFilter because it's not clear what to return in case of false.